File: templates/java/en/text/code/stepUp2.java
/*
 * File: BeeperPickingKarel.java
 * -----------------------------
 * The BeeperPickingKarel program defines a "run" 
 * method with three commands. These commands cause  
 * Karel to move forward one block, pick up a beeper,  
 * and then move ahead to the next corner.
 */
import stanford.karel.*;
public class BeeperPickingKarel extends Karel {
   public void run() {
      move();
      pickBeeper();
      move();
      turnLeft();
      move();
      turnRight();
      move();
      move();
      putBeeper();
      move();
   }

   /**
    * Turns Karel 90 degrees to the right.
    */
   private void turnRight() {
      turnLeft();
      turnLeft();
      turnLeft();
   }
}

File: templates/python/en/text/code/stepUp2.py
# File: BeeperPickingKarel.py
# -----------------------------
# The BeeperPickingKarel program defines a "main" 
# function with three commands. These commands cause  
# Karel to move forward one block, pick up a  
# beeper and then move ahead to the next corner.
from karel.stanfordkarel import *
def main():
   move()
   pick_beeper()
   move()
   turn_left()
   move()
   turn_right()
   move()
   move()
   put_beeper()
   move()

def turn_right():
   turn_left()
   turn_left()
   turn_left()